### ISSUE TITLE
Examples do not pass the test.

### ISSUE THREAD

**Issue by magmax** [Yelp/elastalert#514]
When running `elastalert-test-rule example_rules/example_new_term.yaml  --schema-only`, the result is:
```
elastalert-test-rule example_rules/example_new_term.yaml  --schema-only
Traceback (most recent call last):
  File "/root/tmp/elastalert/venv/bin/elastalert-test-rule", line 9, in <module>
    load_entry_point('elastalert==0.0.80', 'console_scripts', 'elastalert-test-rule')()
  File "/root/tmp/elastalert/venv/local/lib/python2.7/site-packages/elastalert-0.0.80-py2.7.egg/elastalert/test_rule.py", line 287, in main
    test_instance.run_rule_test()
  File "/root/tmp/elastalert/venv/local/lib/python2.7/site-packages/elastalert-0.0.80-py2.7.egg/elastalert/test_rule.py", line 275, in run_rule_test
    hits = self.test_file(copy.deepcopy(rule_yaml), args)
  File "/root/tmp/elastalert/venv/local/lib/python2.7/site-packages/elastalert-0.0.80-py2.7.egg/elastalert/test_rule.py", line 46, in test_file
    load_options(conf, {})
  File "/root/tmp/elastalert/venv/local/lib/python2.7/site-packages/elastalert-0.0.80-py2.7.egg/elastalert/config.py", line 102, in load_options
    raise EAException("Invalid Rule: %s\n%s" % (rule.get('name'), e))
elastalert.util.EAException: Invalid Rule: Example rule
{'index': 'logstash-*', 'terms_window_size': {'days': 90}, 'fields': ['ip_address'], 'alert': ['email'], 'filter': [{'term': {'_type': 'login_event'}}, {'term': {'username': 'admin'}}], 'type': 'new_term', 'email': ['elastalert@ex
ample.com'], 'name': 'Example rule'} is not valid under any of the given schemas

Failed validating 'oneOf' in schema:
    {'$schema': 'http://json-schema.org/draft-04/schema#',
     'definitions': {'arrayOfStrings': {'items': {'type': 'string'},
                                        'type': ['string', 'array']},
                     'arrayOfStringsOrOtherArrays': {'items': {'type': 'array'},
                                                     'type': ['string',
                                                              'array']},
                     'filter': {},
                     'timeFrame': {'additionalProperties': False,
                                   'properties': {'days': {'type': 'number'},
                                                  'hours': {'type': 'number'},
                                                  'milliseconds': {'type': 'number'},
                                                  'minutes': {'type': 'number'},
                                                  'schedule': {'type': 'string'},
                                                  'seconds': {'type': 'number'},
                                                  'weeks': {'type': 'number'}},
                                   'type': 'object'}},
     'oneOf': [{'properties': {'type': {'enum': ['any']}},
                'title': 'Any'},
               {'properties': {'blacklist': {'items': {'type': 'string'},
                                             'type': 'array'},
                               'compare_key': {'type': 'string'},
                               'type': {'enum': ['blacklist']}},
                'required': ['blacklist', 'compare_key'],
                'title': 'Blacklist'},
               {'properties': {'compare_key': {'type': 'string'},
                               'ignore_null': {'type': 'boolean'},
                               'type': {'enum': ['whitelist']},
                               'whitelist': {'items': {'type': 'string'},
                                             'type': 'array'}},
                'required': ['whitelist', 'compare_key', 'ignore_null'],
                'title': 'Whitelist'},
               {'properties': {'compare_key': {'type': 'string'},
                               'ignore_null': {'type': 'boolean'},
                               'timeframe': {'additionalProperties': False,
                                             'properties': {'days': {'type': 'number'},
                                                            'hours': {'type': 'number'},
                                                            'milliseconds': {'type': 'number'},
                                                            'minutes': {'type': 'number'},
                                                            'schedule': {'type': 'string'},
                                                            'seconds': {'type': 'number'},
                                                            'weeks': {'type': 'number'}},
                                             'type': 'object'},
                               'type': {'enum': ['change']}},
                'required': ['query_key', 'compare_key', 'ignore_null'],
                'title': 'Change'},
               {'properties': {'attach_related': {'type': 'boolean'},
                               'doc_type': {'type': 'string'},
                               'num_events': {'type': 'integer'},
                               'terms_size': {'type': 'integer'},
                               'timeframe': {'additionalProperties': False,
                                             'properties': {'days': {'type': 'number'},
                                                            'hours': {'type': 'number'},
                                                            'milliseconds': {'type': 'number'},
                                                            'minutes': {'type': 'number'},
                                                            'schedule': {'type': 'string'},
                                                            'seconds': {'type': 'number'},
                                                            'weeks': {'type': 'number'}},
                                             'type': 'object'},
                               'type': {'enum': ['frequency']},
                               'use_count_query': {'type': 'boolean'},
                               'use_terms_query': {'type': 'boolean'}},
                'required': ['num_events', 'timeframe'],
                'title': 'Frequency'},
               {'properties': {'alert_on_new_data': {'type': 'boolean'},
                               'doc_type': {'type': 'string'},
                               'spike_height': {'type': 'number'},
                               'spike_type': {'enum': ['up',
                                                       'down',
                                                       'both']},
                               'terms_size': {'type': 'integer'},
                               'threshold_cur': {'type': 'integer'},
                               'threshold_ref': {'type': 'integer'},
                               'timeframe': {'additionalProperties': False,
                                             'properties': {'days': {'type': 'number'},
                                                            'hours': {'type': 'number'},
                                                            'milliseconds': {'type': 'number'},
                                                            'minutes': {'type': 'number'},
                                                            'schedule': {'type': 'string'},
                                                            'seconds': {'type': 'number'},
                                                            'weeks': {'type': 'number'}},
                                             'type': 'object'},
                               'type': {'enum': ['spike']},
                               'use_count_query': {'type': 'boolean'},
                               'use_terms_query': {'type': 'boolean'}},
                'required': ['spike_height', 'spike_type', 'timeframe'],
                'title': 'Spike'},
               {'properties': {'doc_type': {'type': 'string'},
                               'threshold': {'type': 'integer'},
                               'timeframe': {'additionalProperties': False,
                                             'properties': {'days': {'type': 'number'},
                                                            'hours': {'type': 'number'},
                                                            'milliseconds': {'type': 'number'},
                                                            'minutes': {'type': 'number'},
                                                            'schedule': {'type': 'string'},
                                                            'seconds': {'type': 'number'},
                                                            'weeks': {'type': 'number'}},
                                             'type': 'object'},
                               'type': {'enum': ['flatline']},
                               'use_count_query': {'type': 'boolean'}},
                'required': ['threshold', 'timeframe'],
                'title': 'Flatline'},
               {'properties': {'alert_on_missing_field': {'type': 'boolean'},
                               'fields': {'items': {'type': 'array'},
                                          'type': ['string', 'array']},
                               'terms_size': {'type': 'integer'},
                               'terms_window_size': {'additionalProperties': False,
                                                     'properties': {'days': {'type': 'number'},
                                                                    'hours': {'type': 'number'},
                                                                    'milliseconds': {'type': 'number'},
                                                                    'minutes': {'type': 'number'},
                                                                    'schedule': {'type': 'string'},
                                                                    'seconds': {'type': 'number'},
                                                                    'weeks': {'type': 'number'}},
                                                     'type': 'object'},
                               'type': {'enum': ['new_term']},
                               'use_terms_query': {'type': 'boolean'}},
                'required': ['fields'],
                'title': 'New Term'},
               {'properties': {'cardinality_field': {'type': 'string'},
                               'max_cardinality': {'type': 'integer'},
                               'min_cardinality': {'type': 'integer'},
                               'timeframe': {'additionalProperties': False,
                                             'properties': {'days': {'type': 'number'},
                                                            'hours': {'type': 'number'},
                                                            'milliseconds': {'type': 'number'},
                                                            'minutes': {'type': 'number'},
                                                            'schedule': {'type': 'string'},
                                                            'seconds': {'type': 'number'},
                                                            'weeks': {'type': 'number'}},
                                             'type': 'object'},
                               'type': {'enum': ['cardinality']}},
                'required': ['cardinality_field', 'timeframe'],
                'title': 'Cardinality'},
               {'properties': {'type': {'pattern': '[.]'}},
                'title': 'Custom Rule from Module'}],
     'properties': {'aggregation': {'additionalProperties': False,
                                    'properties': {'days': {'type': 'number'},
                                                   'hours': {'type': 'number'},
                                                   'milliseconds': {'type': 'number'},
                                                   'minutes': {'type': 'number'},
                                                   'schedule': {'type': 'string'},
                                                   'seconds': {'type': 'number'},
                                                   'weeks': {'type': 'number'}},
                                    'type': 'object'},
                    'alert_text': {'type': 'string'},
                    'alert_text_args': {'items': {'type': 'string'},
                                        'type': 'array'},
                    'alert_text_kw': {'type': 'object'},
                    'alert_text_type': {'enum': ['alert_text_only',
                                                 'exclude_fields']},
                    'buffer_time': {'additionalProperties': False,
                                    'properties': {'days': {'type': 'number'},
                                                   'hours': {'type': 'number'},
                                                   'milliseconds': {'type': 'number'},
                                                   'minutes': {'type': 'number'},
                                                   'schedule': {'type': 'string'},
                                                   'seconds': {'type': 'number'},
                                                   'weeks': {'type': 'number'}},
                                    'type': 'object'},
                    'command': {'items': {'type': 'string'},
                                'type': ['string', 'array']},
                    'email': {'items': {'type': 'string'},
                              'type': ['string', 'array']},
                    'email_reply_to': {'type': 'string'},
                    'es_host': {'type': 'string'},
                    'es_password': {'type': 'string'},
                    'es_port': {'type': 'integer'},
                    'es_username': {'type': 'string'},
                    'exponential_realert': {'additionalProperties': False,
                                            'properties': {'days': {'type': 'number'},
                                                           'hours': {'type': 'number'},
                                                           'milliseconds': {'type': 'number'},
                                                           'minutes': {'type': 'number'},
                                                           'schedule': {'type': 'string'},
                                                           'seconds': {'type': 'number'},
                                                           'weeks': {'type': 'number'}},
                                            'type': 'object'},
                    'field': {},
                    'filter': {'items': {}, 'type': 'array'},
                    'from_addr': {'type': 'string'},
                    'generate_kibana_link': {'type': 'boolean'},
                    'gitter_msg_level': {'enum': ['info', 'error']},
                    'gitter_proxy': {'type': 'string'},
                    'gitter_webhook_url': {'type': 'string'},
                    'hipchat_auth_token': {'type': 'string'},
                    'hipchat_domain': {'type': 'string'},
                    'hipchat_ignore_ssl_errors': {'type': 'boolean'},
                    'hipchat_room_id': {'type': 'number'},
                    'include': {'items': {'type': 'string'},
                                'type': 'array'},
                    'index': {'type': 'string'},
                    'jira_account_file': {'type': 'string'},
                    'jira_bump_tickets': {'type': 'boolean'},
                    'jira_component': {'type': 'string'},
                    'jira_issuetype': {'type': 'string'},
                    'jira_label': {'type': 'string'},
                    'jira_max_age': {'type': 'number'},
                    'jira_project': {'type': 'string'},
                    'jira_server': {'type': 'string'},
                    'kibana_dashboard': {'type': 'string'},
                    'match_enhancements': {'items': {'type': 'string'},
                                           'type': 'array'},
                    'max_query_size': {'type': 'integer'},
                    'name': {'type': 'string'},
                    'notify_email': {'items': {'type': 'string'},
                                     'type': ['string', 'array']},
                    'pagerduty_client_name': {'type': 'string'},
                    'pagerduty_service_key': {'type': 'string'},
                    'pipe_match_json': {'type': 'boolean'},
                    'query_delay': {'additionalProperties': False,
                                    'properties': {'days': {'type': 'number'},
                                                   'hours': {'type': 'number'},
                                                   'milliseconds': {'type': 'number'},
                                                   'minutes': {'type': 'number'},
                                                   'schedule': {'type': 'string'},
                                                   'seconds': {'type': 'number'},
                                                   'weeks': {'type': 'number'}},
                                    'type': 'object'},
                    'query_key': {'items': {'type': 'string'},
                                  'type': ['string', 'array']},
                    'raw_count_keys': {'type': 'boolean'},
                    'realert': {'additionalProperties': False,
                                'properties': {'days': {'type': 'number'},
                                               'hours': {'type': 'number'},
                                               'milliseconds': {'type': 'number'},
                                               'minutes': {'type': 'number'},
                                               'schedule': {'type': 'string'},
                                               'seconds': {'type': 'number'},
                                               'weeks': {'type': 'number'}},
                                'type': 'object'},
                    'slack_emoji_override': {'type': 'string'},
                    'slack_msg_color': {'enum': ['good',
                                                 'warning',
                                                 'danger']},
                    'slack_username_override': {'type': 'string'},
                    'slack_webhook_url': {'type': 'string'},
                    'smtp_host': {'type': 'string'},
                    'telegram_api_url': {'type': 'string'},
                    'telegram_bot_token': {'type': 'string'},
                    'telegram_room_id': {'type': 'string'},
                    'timestamp_field': {'type': 'string'},
                    'top_count_keys': {'items': {'type': 'string'},
                                       'type': 'array'},
                    'top_count_number': {'type': 'integer'},
                    'use_kibana_dashboard': {'type': 'string'},
                    'use_local_time': {'type': 'boolean'},
                    'use_ssl': {'type': 'boolean'},
                    'use_strftime_index': {'type': 'boolean'},
                    'victorops_api_key': {'type': 'string'},
                    'victorops_entity_display_name': {'type': 'string'},
                    'victorops_message_type': {'enum': ['INFO',
                                                        'WARNING',
                                                        'ACKNOWLEDGEMENT',
                                                        'CRITICAL',
                                                        'RECOVERY']},
                    'victorops_routing_key': {'type': 'string'}},
     'required': ['name', 'type', 'index', 'alert'],
     'type': 'object'}

On instance:
    {'alert': ['email'],
     'email': ['elastalert@example.com'],
     'fields': ['ip_address'],
     'filter': [{'term': {'_type': 'login_event'}},
                {'term': {'username': 'admin'}}],
     'index': 'logstash-*',
     'name': 'Example rule',
     'terms_window_size': {'days': 90},
     'type': 'new_term'}
```

Examples should be tested in travis in order to ensure their validity (at least the schema)

**Comment by Qmando** [comment]
You are right, Travis should be testing these. This was introduced in #473 and just fixed by #488

**Comment by xiaobao2** [comment]
+1
